testing/testcafe/tests/editors/overlays/resizeObserverIntegration.ts

```diff
@@ -204,33 +198,37 @@ test('Popup dimensions should be correct after width or height animation', async
 test('Showing and shown events should be raised only once even after resize during animation', async (t) => {
   const popup = new Popup('#container');
 
-  await popup.show();
-  await setStyleAttribute(Selector('#content'), 'width: 300px; height: 300px;');
+  await ClientFunction(() => {
+    (window as any).shownCallCount = 0;
+    (window as any).showingCallCount = 0;
+  })();
 
-  const shownCallCount = Number(await Selector('#shown_call_count').innerText);
-  const showingCallCount = Number(await Selector('#showing_call_count').innerText);
+  const incShown = ClientFunction(() => { ((window as any).shownCallCount as number) += 1; });
+  const incShowing = ClientFunction(() => { ((window as any).showingCallCount as number) += 1; });
+
+  const getShownCounter = ClientFunction(() => (window as any).shownCallCount);
+  const getShowingCounter = ClientFunction(() => (window as any).shownCallCount);
+
+  await popup.option({
+    onShown: incShown,
+    onShowing: incShowing,
+  });
+
+  await popup.show();
 
   await t
-    .expect(shownCallCount)
+    .expect(await getShownCounter())
     .eql(1);
   await t
-    .expect(showingCallCount)
+    .expect(await getShowingCounter())
     .eql(1);
-}).before(async () => {
-  await appendElementTo('body', 'div', 'shown_call_count', {});
-  await appendElementTo('body', 'div', 'showing_call_count', {});
-
-  return createWidget('dxPopup', {
-    width: 'auto',
-    height: 'auto',
-    contentTemplate: () => $('<div>').attr({ id: 'content' }).css({ width: 100, height: 100 }),
-    onShown: ClientFunction(() => {
-      const callCount = +$('#shown_call_count').text();
-      $('#shown_call_count').text(callCount + 1);
-    }),
-    onShowing: ClientFunction(() => {
-      const callCount = +$('#showing_call_count').text();
-      $('#showing_call_count').text(callCount + 1);
-    }),
-  });
+}).before(async () => createWidget('dxPopup', {
+  width: 'auto',
+  height: 'auto',
+  contentTemplate: () => $('<div>').attr({ id: 'content' }).css({ width: 100, height: 100 }),
+})).after(async () => {
+  await ClientFunction(() => {
+    (window as any).shownCallCount = 0;
```

Comment:
I think we should delete it to clear memory, not just set to `0`.
By the way, y do not we use `ctx` here?

Reply:
We can't get access to ctx here, because it uses in client code.